### PR TITLE
Test test test

### PR DESCRIPTION
**Pull request by ena-fly**
Added tests for lights and reliability success rate test.
Working with lights v146.

### CODE REVIEW

File: beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt [hunk 1]
```diff
@@ -56,7 +56,7 @@ sealed interface Test {
 
     data class Action(val execute: suspend () -> Either<TestFailed, Unit>) : Test
 
-    data class Repeat(val times: Int, val test: Test) : Test
```
Comment (lenguyenthanh) [inline beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt]: why `SuccessRate` here? The `Repeat` is supposed to repeat any kind of tests?

File: beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        val message: SendVendorModelMessage,
+        val assert: Assertion<ProxyFilterMessage>,
+    ) : Test
+
+    data class AssertOnly(
+        val expectedSize: Int,
+        val assert: Assertion<List<ProxyFilterMessage>>
+    ) : Test
+
+    data class MultipleVendorMessage(
```
Comment (lenguyenthanh) [inline beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt:53]: This is not `MultipleVendorMessage` anymore. Should we move it to `MultipleTests`?

**Review by lenguyenthanh — CHANGES_REQUESTED** [review]

File: beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt (hunk 1 above)
Comment (ena-fly) [reply beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt]: We are doing SuccessRate test but we are only testing on 2 lights so I just wanted to repeat that couple of times so we can see the successRate after multiple calls

Comment (ena-fly) [reply beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt]: Oh yes, I need to change it back to Test. My mistake, fixing it 👍

File: beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+}
+
+sealed interface TestFailed {
+    data class ExecutionError(val error: Any?) : TestFailed
+    data class NotEqual<T>(val test: Test, val actual: T) : TestFailed
+    data class MultipleFailed(val fails: Nel<TestFailed>) : TestFailed
+    object NotInConnectedSTate : TestFailed
+}
+
+sealed interface MqttConnectError : TestFailed {
```
Comment (lenguyenthanh) [inline beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt:196]: Beckon shouldn't have any knowledge about Mqtt. We should have something like:
```
   class ActionFailed(throwable: Throwable): TestFailed
```

File: beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    data class AssertOnly(
+        val expectedSize: Int,
+        val assert: Assertion<List<ProxyFilterMessage>>
+    ) : Test
+
+    data class MultipleVendorMessage(
+        val vendorMessages: List<Test>
+    ) : Test
+
+    data class Action(val execute: suspend () -> Either<TestFailed, Unit>) : Test
```
Comment (lenguyenthanh) [inline beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt:57]: `Either<TestFailed, Unit>` => `Either<ActionFailed, Unit>`

**Comment by lenguyenthanh** [comment]
Great job @ena-fly !!! Can you also rebase it to main?

**Review by lenguyenthanh — APPROVED** [review]

**Pull request merged**